recipes-kernel/linux/5.4/patches/blktap2.patch

```diff
@@ -791,7 +757,10 @@ PATCHES
 +		"end_request: op=%d error=%d bytes=%d\n",
 +		rq_data_dir(rq), error, blk_rq_bytes(rq));
 +
-+	blktap_end_rq(rq, error);
++	// XXX Maybe this lock is unneeded
++	spin_lock_irq(&rq->q->queue_lock);
```

Comment:
Curious why you might think the lock is needed? I haven't traced the logic here in blktap2 to know if there might be a race condition but I looked at a few other block device drivers using blk_mq and after a half dozen or so I found none was protecting their call to `blk_mq_end_request` with a spinlock.

Reply:
The locking already existed and moved with inlining blktap_end_rq.  I kept it, but like you found, it may not be needed.  But I didn't thoroughly investigate, so I left it.

Reply:
Understood. It doesn't hurt anything but consuming a few cycles to lock and unlock but doubt that is causing any noticeable performance impacts. I have no issue with it staying and if someone wants to step up to confirm its not needed at which point it can then get dropped.